### DISCUSSION TITLE
Make script smaller

### DISCUSSION THREAD

**Discussion by Jak2k** [bigskysoftware/htmx#1996] [Q&A]
For my app, every byte matters. Is it possible to bundle a htmx-script with just a small subset of features?

**Comment by Telroshan** [comment]
Hey, I think this has already been requested a few times (but can't find where again, maybe on the Discord); this isn't planned short term _(nor mid term really)_, so I wouldn't advise you to wait for an official release of that kind anytime soon!
Feel free to make your own though! I recall [laser.js](https://gist.github.com/alexbezhan/59e86e77315baff24fd161ff1a588947) from the top of my head but I'm sure I had seen other pass by on the discord. Check them out, they might fit your needs :) otherwise you'd probably want to make your own tailored htmx, stripping all the features you don't use; if every byte really matters, you would probably end up doing this anyway!

**Reply by niutech** [reply]
Htmx 1.0.2 was [27KB minified](https://github.com/bigskysoftware/htmx/blob/v1.0.2/dist/htmx.min.js) and htmx 1.9.11 is whooping [47KB minified](https://github.com/bigskysoftware/htmx/blob/v1.9.11/dist/htmx.min.js). This is 74% increase in size. Why can't you remain small size?

**Reply by nickchomey** [reply]
How is this even a question? 

Because people - the maintainers in particular - want features.

There's other libraries out there if this doesn't serve your needs...

**Reply by niutech** [reply]
Not every feature should be included, otherwise it would become bloated like Angular in a year or so. KISS 💋

**Reply by nickchomey** [reply]
Go elsewhere

**Reply by niutech** [reply]
@nickchomey Don't be rude. This thread is raising a valid concern. From being lightweight at the beginning, htmx is getting more and more bloated with every minor version. It's time to stop it.

**Reply by Telroshan** [reply]
> Not every feature should be included, otherwise it would become bloated like Angular in a year or so. KISS 💋

Your point is valid, and it's why we have [htmx extensions](https://htmx.org/extensions/#included).
As you said, not every feature should be included, which is also why, as per our [contribution guidelines](https://github.com/bigskysoftware/htmx/blob/master/CONTRIBUTING.md)
> When proposing a new feature, we will often suggest that you implement it as an extension, so try that first

It can be tough to find the right balance between the core features that the average user would need, and the ones that should be extensions. As a single individual, it can be easy to fall into the wrong assumption that your own needs are everyone's needs ; features that seem core to some people might seem completely optional to others, so goes the other way around, it's then a matter of compromises.

Htmx 1.9.11 is certainly heavier than htmx 1.0.2 but includes many improvements, bugfixes, and features that _felt_ core, and were requested by the community.

We all have our own opinions, please keep this a "good vibes only" place on the internet.

If every byte matters for you, as said above, I encourage you to fork htmx and remove the parts you don't care about. That's what I'm personally doing for features I simply "don't like" or don't care about, and that feels like a perfectly reasonable approach to me regarding open source ; you have access to everything, complaining is easy, yet contributing and making your fork are also easy.
Htmx is a single file lib, so if you really need that weight optimization, you have all the tools available to suit your very specific usecase.
Please respect that the average htmx newcomer might need a broader set of default features to have by hand, before looking into extensions for more precise usecases.

**Reply by Jak2k** [reply]
Maybe it would be possible to make a site, where you can select the features you want/need and then you get you custom htmx. I'll add that to my todos…

**Reply by nickchomey** [reply]
@Jak2k thats what es modules are for. I believe the project explicitly decided not to go that route, which is their prerogative. 

Datastar is one library that im aware of that does similar things to htmx but more modularized.

**Reply by niutech** [reply]
[Datastar](https://datastar.fly.dev) is a lightweight alternative, weighting 30KB.

**Comment by niutech** [comment]
For a much smaller library, check out [htmz](https://leanrada.com/htmz/).

**Reply by Jak2k** [reply]
That's a bit too small 😄

**Reply by niutech** [reply]
@Jak2k Less is more, YAGNI 😃
